lib/src/http/http_body.dart

```diff
@@ -132,10 +131,11 @@ class HttpBody extends BaseHttpBody {
   /// Returns this request/response body as a JSON object - either a `Map` or a
   /// `List`.
   ///
-  /// This attempts to read this request/response body as a `String` and decode
-  /// it to a JSON object. Throws a [FormatException] if this request/response
-  /// body cannot be decoded to text or if the text is not valid JSON.
-  dynamic asJson() => json.decode(asString());
+  /// This attempts to read this request/response body as a UTF-8 String and
+  /// decode it to a JSON object. Throws a [FormatException] if this
+  /// request/response body cannot be decoded to text or if the text is not
+  /// valid JSON.
+  dynamic asJson() => json.decode(utf8.decode(asBytes()));
```

Comment:
Possibly this should be use the encoding if it was specified and otherwise utf-8, but my impression is that it's pretty much always utf-8. Like, the HTTP spec says Accept-Charset is deprecated because there's no point.